### ISSUE TITLE
Eslint Plugin Prefer Let

### ISSUE THREAD

**Issue by jorgegonzalez** [xojs/xo#307]
What does XO think about considering `eslint-plugin-prefer-let`? https://www.npmjs.com/package/eslint-plugin-prefer-let

> Usage of the const keyword to bind an intermediate value of a computation places emphasis on the compiler and and its role in ensuring that a reference never changes. By contrast using let in the same situation reads, in plain English, the programmer's intent to declare a name value binding.

> It is this plugin's opinion that preventing reassignment of let bindings is better accomplished as a linting rule.

I know this is typically not thought of as a "best practice", but it is an interesting consideration and I thought it might be a happy addition to XO ;)

**Comment by pvdlg** [comment]
Thanks for the suggestion.

This plugin and the rule it enforces seems very very opinionated and far from being something widely accepted as a good practice. Therefore we most likely won't add it in XO.

As you notice you can use it via configuration. As mentioned in the plugin documentation you have to use the plugin and the rule:
```json
"xo": {
    "plugins": ["prefer-let"],
    "rules": {
        "prefer-let/prefer-let": 2
    }
}
```

**Comment by sindresorhus** [comment]
You could also add it to a shareable ESLint config and use the `extends` option.

**Comment by jorgegonzalez** [comment]
Thank you @pvdlg and @sindresorhus

**Comment by jorgegonzalez** [comment]
Hmm `[XO] 'maxLength' is never reassigned. Use 'const' instead. (prefer-const)` seems to overrule the plugin.

**Comment by pvdlg** [comment]
That's mostly likely an issue to report on `eslint-plugin-prefer-let`. If the plugin has conflicts with other eslint rules, they should mention it in the doc so users could disable the problematic rules.

**Comment by jorgegonzalez** [comment]
You're right @pvdlg, had to set this:

```
"rules": {
  "prefer-let/prefer-let": 2,
	"prefer-const": "off"
}
```

Thanks for the help.
